pkg/auth/token.go

```diff
@@ -0,0 +1,62 @@
+package auth
+
+import (
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/fastly/cli/pkg/api"
+	"github.com/fastly/cli/pkg/config"
+	"github.com/fastly/cli/pkg/text"
+	"github.com/fastly/go-fastly/v6/fastly"
+)
+
+// ValidateToken retrieves the given token data and validates if it's OK to
+// continue using.
+func ValidateToken(apiClient api.Interface, stdout io.Writer) (ok bool) {
+	t, err := apiClient.GetTokenSelf()
+	if err != nil {
+		text.Info(stdout, "We were not able to validate your access token.")
+		return false
+	}
+
+	if t.ExpiresAt == nil {
+		text.Info(stdout, "Your current access token has no expiration and will be replaced with a short-lived token.")
+		return false
+	}
+
+	if (*t.ExpiresAt).Before(time.Now()) {
+		text.Info(stdout, "Your access token has expired.")
+		return false
+	}
+
+	return true
+}
```

Comment:
Here is where we validate a pre-existing token.

The token is read from the CLI config file and is going to be either a long-lived token (added by the user via `fastly configure` before the CLI supported OAuth) or a short-lived token generated by the new authentication flow.